src/Sql/dbo/Stored Procedures/User_ReadBySsoUserOrganizationIdExternalId.sql

```diff
@@ -0,0 +1,20 @@
+﻿CREATE PROCEDURE [dbo].[User_ReadBySsoUserOrganizationIdExternalId]
+    @OrganizationId UNIQUEIDENTIFIER,
+    @ExternalId NVARCHAR(50)
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        U.*
+    FROM
+        [dbo].[UserView] U
+    INNER JOIN
+        [dbo].[SsoUser] SU ON SU.[UserId] = U.[Id]
+    WHERE
+        (
+            (@OrganizationId IS NULL AND SU.[OrganizationId] IS NULL)
```

Comment:
During SSO login we only have an External ID (providerUserId) during lookup, so if the matching SsoUser has an Organization ID, wouldn't that fail this condition?

Reply:
Wouldn't we also have the organization's identifier/external ID? You would use that to lookup the org, get the org id ('cause that's how you got the SSO profile) and pass that in here right?

Reply:
That's true, we'll have an org ID from the SsoConfig lookup when establishing a domainHint.  Is it ok to include the ID as an Item within AuthenticationProperties to avoid doing another lookup?

Reply:
Yes, we can pass it along through other channels to keep state.